src/components/Navigation/Navigation.jsx

```diff
@@ -0,0 +1,59 @@
+import AddIcon from '@mui/icons-material/AddCircleOutlineOutlined';
+import ListIcon from '@mui/icons-material/FormatListBulletedOutlined';
+import LogoutIcon from '@mui/icons-material/LogoutOutlined';
+import ShareIcon from '@mui/icons-material/ShareOutlined';
+import { IconButton } from '@mui/material';
+import { Link } from 'react-router-dom';
+import Title from '../../assets/smartshopperr_title_transparent.png';
+import './Navigation.css';
+
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+
+export const Navigation = ({ setToken, token }) => {
+  const navigate = useNavigate();
+
+  const handleSignOut = () => {
+    setToken('');
+    localStorage.removeItem('token');
+    navigate('/');
+  };
+
+  const copyToClipboard = () => {
+    navigator.clipboard.writeText(token).then(() =>
+      toast('Token copied to clipboard!', {
+        autoClose: 3000,
+        hideProgressBar: true,
+        pauseOnHover: false,
+        draggable: false,
+      }),
+    );
+  };
+
```

Comment:
For future development, I wonder if we can also display the token? Modal? Toast?

Reply:
For clarification, display it when copied or displayed consistently on the page? I think it's a great suggestion, though - it adds more transparency for the user!

Reply:
When copied. What do you think?

Reply:
Figured out a way to add it to the toast notification. It was fun figuring it out, so thank you for the suggestion!

Reply:
This is exciting!